shared/System.IO.Pipelines.Extensions.Sorces/BufferReader.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+
+namespace System.IO.Pipelines.Extensions
+{
+    public ref struct BufferReader
```

Comment:
Public types in a `.Sources` package? Is this on purpose?

Reply:
Should be internal.

Reply:
Guys, It's a `Sorces` package. It's fine

Reply:
lol